### DISCUSSION TITLE
p2p: Increase inbound capacity for block-relay only connections

### DISCUSSION THREAD

**Discussion by Prabhat1308** [Bitshala/BitcoinCore-PR-Review-Club#82] [PR Reviews]
## Session Details
Date:22-08-2024
Time: IST 20:00 (UTC 14:30)
Link : https://github.com/bitcoin/bitcoin/pull/28463
Difficulty : Medium

## Notes
This PR propose to increase default maximum connections from 125 to 200, with 50% of inbound slots accessible for tx-relaying peers. 
This is a prerequisite for being able to increase the default number of outgoing block-relay-only peers later, because the current inbound capacity of the network is not sufficient.
In order to account for incoming tx-relaying peers separately from incoming block-relay peers, changes to the inbound eviction logic are necessary.

## Questions
1. What are feeler connections ? What is their utility ?
2. What are the conditions / requirements for the evictions of the peers when peers are being selected to be evicted .
3. How is the number of inbound tx-relay connections being maintained after the increase of nodes ?

## Learning
P2P 
tests

**Comment by QureshiFaisal** [comment]
Ans 1. A feeler connection is a temporary outbound connection initiated by your node after it has already established the necessary 8 outbound full-relay connections and 2 block-relay-only connections. The main purpose of a feeler connection is to periodically test addresses from the "new" table to determine if they are reachable Bitcoin nodes.

Ans 2. Some of the conditions on the basis of which a node can be evicted are as follows:

-  If a node provides invalid data, such as a mutated block or data that doesn’t meet consensus requirements, it will  be disconnected.
- If a peer fails to send a version message as the first message, or if it sends more than one version message per connection
-  If a peer sends an inv message exceeding the maximum size.
- Sending addr messages with more than 1,000 addresses.
All of the above mentioned reasons increase the ban-score of a peer.
However, in context to this PR, there is an additional criteria i.e the peer should be a tx-relay-only peer and the number of tx-relay blocks should have reached the upper bound.

Ans 3. The number of inbound tx-relay connections in context to this PR is being capped at 50% of the total automatic inbound connections, which is 95 peers.

**Comment by paperpsych** [comment]
@VirdiJatin 
![Bitshala RC (28463)](https://github.com/user-attachments/assets/0df5b482-0cfd-4502-b4e6-6b034187ff02)
